exp/addons/internal/controllers/clusterresourcesetbinding_controller.go

```diff
@@ -77,8 +78,33 @@ func (r *ClusterResourceSetBindingReconciler) Reconcile(ctx context.Context, req
 		// Error reading the object - requeue the request.
 		return ctrl.Result{}, err
 	}
-
-	cluster, err := util.GetOwnerCluster(ctx, r.Client, binding.ObjectMeta)
+	// Before 1.4 cluster name was stored as ownerRef; this code migrated this value to
+	// the spec.clusterName field if necessary.
+	clusterName := binding.Spec.ClusterName
+	if clusterName == nil {
+		// Initialize the patch helper.
+		patchHelper, err := patch.NewHelper(binding, r.Client)
+		if err != nil {
+			log.Error(err, "Failed to configure the patch helper")
+			return ctrl.Result{Requeue: true}, nil
+		}
+		// Update the clusterName field of the existing ClusterResourceSetBindings with ownerReferences.
+		// More details please refer to: https://github.com/kubernetes-sigs/cluster-api/issues/7669.
+		if clusterName, err = getClusterNameFromOwnerRef(binding.ObjectMeta); err != nil {
+			return ctrl.Result{}, err
+		} else if clusterName == nil {
+			log.Info("ownerRef not found for the ClusterResourceSetBinding", "ClusterResourceSetBindingName", binding.Name)
```

Comment:
```suggestion
			log.Info("ownerRef not found for the ClusterResourceSetBinding")
```
Not necessary. Controller-runtime already adds name and namespcae of the ClusterResourceSetBinding as k/v pair

Reply:
Q: Shouldn't we return this as error instead in l.97?

Reply:
> Not necessary. Controller-runtime already adds name and namespcae of the ClusterResourceSetBinding as k/v pa

ok, I'm having Chinese New Year, I'll fix this when I get back from the holidays.

Reply:
Sure, absolutely no rush!